components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/streaming/websocket/WebSocketAnalyticsDataProvider.java

```diff
@@ -0,0 +1,283 @@
+        return isAuthenticated() && APIConstants.END_USER_ANONYMOUS.equalsIgnoreCase(authContext.getUsername());
+    }
+
+    @Override
+    public boolean isAuthenticated() {
+        AuthenticationContext authContext = getAuthenticationContext();
+        return authContext != null && authContext.isAuthenticated();
+    }
+
+    @Override
+    public FaultCategory getFaultType() {
+        if (isAuthFaultRequest()) {
+            return FaultCategory.AUTH;
+        } else if (isThrottledFaultRequest()) {
+            return FaultCategory.THROTTLED;
+        } else if (isTargetFaultRequest()) {
+            return FaultCategory.TARGET_CONNECTIVITY;
+        }
+        return FaultCategory.OTHER;
+    }
+
+    private boolean isAuthFaultRequest() {
+        int errorCode = getErrorCode();
+        return errorCode >= Constants.ERROR_CODE_RANGES.AUTH_FAILURE_START
+                && errorCode < Constants.ERROR_CODE_RANGES.AUTH_FAILURE__END;
+    }
+
+    private boolean isThrottledFaultRequest() {
+        int errorCode = getErrorCode();
+        return errorCode >= Constants.ERROR_CODE_RANGES.THROTTLED_FAILURE_START
+                && errorCode < Constants.ERROR_CODE_RANGES.THROTTLED_FAILURE__END;
+    }
+
+    private boolean isTargetFaultRequest() {
+        return false;
+    }
+
+    @Override
+    public API getApi() {
+        String apiContext = (String) WebSocketUtils.getPropertyFromChannel(RESTConstants.REST_API_CONTEXT, ctx);
+        String apiVersion = (String) WebSocketUtils.getPropertyFromChannel(RESTConstants.SYNAPSE_REST_API_VERSION, ctx);
+        String tenantDomain = MultitenantUtils.getTenantDomainFromRequestURL(apiContext);
+        if (tenantDomain == null) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        SubscriptionDataStore store = SubscriptionDataHolder.getInstance().getTenantSubscriptionStore(tenantDomain);
+        org.wso2.carbon.apimgt.keymgt.model.entity.API apiObj = store.getApiByContextAndVersion(apiContext, apiVersion);
+        API api = new API();
+        if (apiObj == null) {
+            try {
+                apiObj = new SubscriptionDataLoaderImpl().getApi(apiContext, apiVersion);
+            } catch (DataLoadingException e) {
+                log.error("Error occurred when getting api.", e);
+                throw new RuntimeException("Error occurred when getting API information", e);
+            }
+        }
+        if(apiObj != null) {
+            api.setApiId(apiObj.getUuid());
+            api.setApiType(apiObj.getApiType());
+            api.setApiName(apiObj.getApiName());
+            api.setApiVersion(apiObj.getApiVersion());
+            api.setApiCreator(apiObj.getApiProvider());
+            api.setApiCreatorTenantDomain(MultitenantUtils.getTenantDomain(api.getApiCreator()));
+        }
+        return api;
+    }
+
+    @Override
+    public Application getApplication() throws DataNotFoundException {
```

Comment:
cant we use the super?

Reply:
We are implementing the `AnalyticsDataProvider` interface here (not the `SynapseAnalyticsDataProvider` class), therefore we can't use super.

Reply:
+1